internal/config/config.go

```diff
@@ -41,28 +42,38 @@ type cfg struct {
 	cfg *ghConfig.Config
 }
 
-func (c *cfg) Get(hostname, key string) (string, error) {
+func (c *cfg) get(hostname, key string) o.Option[string] {
 	if hostname != "" {
 		val, err := c.cfg.Get([]string{hostsKey, hostname, key})
 		if err == nil {
-			return val, err
+			return o.Some(val)
 		}
 	}
 
-	return c.cfg.Get([]string{key})
+	val, err := c.cfg.Get([]string{key})
+	if err == nil {
+		return o.Some(val)
+	}
+
+	return o.None[string]()
 }
 
-func (c *cfg) GetOrDefault(hostname, key string) (string, error) {
-	val, err := c.Get(hostname, key)
-	if err == nil {
-		return val, err
+func (c *cfg) GetOrDefault(hostname, key string) o.Option[gh.ConfigEntry] {
+	if val := c.get(hostname, key); val.IsSome() {
+		return o.Map(val, toConfigEntry(gh.ConfigUserProvided))
```

Comment:
This is probably the most functional programming suspicious line. `o.Map` is a function that takes an `Optional` and provides the inner value to a function, and wraps the result of that function in an optional. This means we can take an `Option[string]` and map it to `Option[ConfigEntry]` without the manual process of unwrapping the value and creating a new `Some` ourselves.

You can think of it like `Array.map` in other languages, where the function is applies to each element of an array, and they are all wrapped up in a new array, except that it's a single element.

Reply:
It sounds like a comment in this section of code might be helpful for future readers

Reply:
Sounds like a great idea.